### DISCUSSION TITLE
Retries on finalization after watcher reconnect

### DISCUSSION THREAD

**Discussion by anekdoti** [dotnet/dotnet-operator-sdk#304] [Q&A] [answered]
Finalization is only attempted a fixed number of times. But when I understand the KubeOps source code right, the event that is triggered for the (not successfully finalized resource) after a watcher reconnect should also start an attempt for finalization again (as long as the DeletionTimestamp is set on the resource). I.e., in practice, finalization is retried an unlimited number of times.

Could anybody confirm this or is there a caveat I am missing?

**Comment by erin-allison** [comment, marked as answer]
I don't have the time to dig through the code this morning, but from memory, your understanding sounds correct.

**Reply by buehler** [reply]
finalization is retried an unlimited number of times. since everytime the resource is compared, it has a deleted-on timestamp as well as registered finalizers.

**Reply by anekdoti** [reply]
Thank you very much for your replies!
